### ISSUE TITLE
Possible bug in allow_dask_on_strings

### ISSUE THREAD

**Issue by letfoolsdie** [jmcarpenter2/swifter#53]
https://github.com/jmcarpenter2/swifter/blob/3b693c1977c1d67951874e756043be1947c80fe7/swifter/swifter.py#L146

Shouldn't it be `if self._allow_dask_on_strings ` instead of `if not self._allow_dask_on_strings`?
(and the same in DataFrameAccessor)

Currently it looks like if we set `self._allow_dask_on_strings` to True and pass string series, `str_processing` will be False which is not we want probably.

**Comment by jmcarpenter2** [comment]
Hey @letfoolsdie,

Thanks for looking through the package and seeking out bugs! It's always good to get a second eye on these things.

When the PR came in that implemented this functionality, I had to take a second look at this logic as well. 

The way this logic reads is like so

```
if self._allow_dask_on_strings == True:
    return False  # Allows dask processing by setting this variable to False
else:
    if samp_dtype == "object"
         return True   # Does not allow dask processing
    else:
         return False  # Allows dask processing
```

I think the best solution here is to rename the `str_processing` variable to something that is more reflective of what the variable actually does and probably remove all the double-negative logic here. The reason for the poorly named variable is that I originally made it just to check if we were doing text processing but when the PR came in to allow `allow_dask_on_strings` then I forgot to also change the variable name to something more apt.

Thanks for highlighting this lack of clarity, I will submit a PR to fix this up sometime this week.
Jason

**Comment by jmcarpenter2** [comment]
Fixed in [#55](https://github.com/jmcarpenter2/swifter/pull/55)
